patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -5827,6 +5928,31 @@
  			try {
  				RijndaelManaged rijndaelManaged = new RijndaelManaged();
  				rijndaelManaged.Padding = PaddingMode.None;
+@@ -38410,16 +_,24 @@
+ 
+ 							player.statLife = binaryReader.ReadInt32();
+ 							player.statLifeMax = binaryReader.ReadInt32();
++							// Allow health stat to exceed vanilla bounds (#HealthManaAPI)
++							/*
+ 							if (player.statLifeMax > 500)
+ 								player.statLifeMax = 500;
++							*/
+ 
+ 							player.statMana = binaryReader.ReadInt32();
+ 							player.statManaMax = binaryReader.ReadInt32();
++							// Allow mana stat to exceed vanilla bounds (#HealthManaAPI)
++							/*
+ 							if (player.statManaMax > 200)
+ 								player.statManaMax = 200;
+ 
+ 							if (player.statMana > 400)
+ 								player.statMana = 400;
++							*/
++
++							PlayerLoader.ModifyMaxStats(player);
```

Comment:
why this call? `ModPlayer` data hasn't been loaded yet at this point

Reply:
In fact, this could mess with the vanilla fallback `ConsumedX` calculations if a mod just, for eg, adds a blanket 100hp.

Why not set the `ConsumedX` properties here, instead of using tML NBT data at all, since you'll have the vanilla data in all cases as that's what was saved.

Reply:
good point